File: greatest-common-divisor-euclidean-algorithm/gcd.java
class gcd {
  public static int gcd( int num1 , int num2 ) {
    if ( num1 < num2 ) return gcd( num2 , num1 ) ;
    int rem = num1 % num2 ;
    return ( rem == 0 ) ? num2 : gcd( num2 , rem ) ;
  }
}

File: greatest-common-divisor-euclidean-algorithm/gcd.kt
fun gcd( num1 : Int , num2 : Int ) : Int {
  if ( num1 < num2 ) return gcd( num2 , num1 ) 
  val rem : Int = num1 % num2 
  return if ( rem == 0 ) num2 
  else gcd( num2 , rem )
}

fun main() {
  println( gcd( 30 , 9 )) 
  println( gcd( 100 , 10 )) 
}
